bin/main.rb

```diff
@@ -1,67 +1,86 @@
+
 class Board
   attr_accessor :board
 
   def initialize
     @board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    @count = 1
+    @game_status = GameStatus.new
   end
 
   def display_board
-    puts "#{board[0]} | #{board[1]} | #{board[2]}\n---------\n#{board[3]} | #{board[4]} | #{board[5]}\n---------
-    \n#{board[6]} | #{board[7]} | #{board[8]}"
+    puts "#{@board[0]} | #{@board[1]} | #{@board[2]}"
+    puts '---------'
+    puts "#{@board[3]} | #{@board[4]} | #{@board[5]}"
+    puts '---------'
+    puts "#{@board[6]} | #{@board[7]} | #{@board[8]}"
   end
-end
 
-class Player
-  attr_reader :name, :sign
-  attr_accessor :record
+  def welcome
+    puts 'Welcome to Tic Tac Toe'
+    puts ''
+  end
 
-  def initialize(name, sign)
-    @name = name
-    @sign = sign
-    @record = []
+  def player_setup
+    puts 'Player X please input your name'
+    name = gets.strip
+    @player1 = Player.new(name, 'X')
+    puts 'Player O please input your name'
+    name = gets.strip
+    until name != @player1.name
+      puts 'please input a different name, that name is already taken'
+      name = gets.strip
+    end
+    @player2 = Player.new(name, 'O')
   end
-end
 
-def win_check(win_combination, player_record)
-  win_combination.each do |win|
-    win.all? { |win1| player_record.include?(win1) }
+  def current_player
+    if @count.odd?
+      @player1
+    else
+      @player2
+    end
   end
-end
 
-puts 'Player 1 please input your name'
-name = gets.strip
-token = 'X'
-Player_1 = Player.new(name, token)
-puts 'Player 2 please input your name'
-name = gets.strip
-token = 'O'
-Player_2 = Player.new(name, token)
-game_board = Board.new
-# win_combination = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]]
-count = 1
-game_on = true
-while game_on
-  puts game_board.display_board
-  current_player = if count.odd?
-                     Player_1
-                   else
-                     Player_2
-                   end
-  puts "#{current_player.name} it is your turn, please make your move"
-  move = gets.strip
-  gamecheck = game_board.board[move.to_i - 1]
-  puts 'Move is not valid, Make a valid move' unless move.to_i == gamecheck
-  current_player.record << move.to_i
-  game_board.board[move.to_i - 1] = current_player.sign
-  #   if win_check(win_combination, current_player.record)
-  #     puts '#{current_player.name} is the winner for this round'
-  #     game_on = false
-  #   end
-  count += 1
-  if count == 10
-    puts 'The game is a draw'
-    break
+  def turn
+    @game_over = false
+
+    puts "#{current_player.name} it is your turn, please make your move choosing between 1 to 9"
+    input_board
+    display_board
+    if @game_status.win_check(@board)
+      puts "#{current_player.name} is the winner"
+      return @game_over = true
+    end
+
+    if @game_status.draw?(@count)
+      puts 'The game is a draw'
+      return @game_over = true
+    end
+    @count += 1
+    current_player
+  end
```

Comment:
- [ ] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). don't create long methods that are doing multiple steps of the game process. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/single_responsibility.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- The code within this method seems to belong directly in the main.rb file within a loop and not in a method because it checks many things so the responsibility is unclear and it could be very hard to test for a future project later on so please keep the contents of this method in the `main.rb` file within some sort of loop